### PR TITLE
Fixes pangeo-data/pangeo #176

### PR DESCRIPTION
**Pull request by martindurant**
Google apparently changed their error message when a key does not exist.

### CODE REVIEW

File: gcsfs/core.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def validate_response(r, path):
         except:
             msg = str(r.content)
 
-        if "Not Found" in m:
+        if "Not Found" in m or 'No such object' in m:
             raise FileNotFoundError(path)
```
Comment (mrocklin) [inline gcsfs/core.py]: Is there any way to make these checks less fragile?  For example is there a status code that is both sufficiently meaningful and less likely for google to change?

Comment (martindurant) [reply gcsfs/core.py]: According to https://cloud.google.com/storage/docs/json_api/v1/status-codes , yes.

**Comment by rabernat** [comment]
Thanks @martindurant for the quick fix!

**Comment by mrocklin** [comment]
@rabernat are you able to verify that this resolves the issue?

**Comment by mrocklin** [comment]
If so then I'll push a new image up

**Comment by rabernat** [comment]
> @rabernat are you able to verify that this resolves the issue?

Yes, it resolves the issue. 👍

**Pull request merged**